shell/shell.js

```diff
@@ -16,22 +16,153 @@
 
 const { classes: Cc, interfaces: Ci, results: Cr, utils: Cu } = Components;
 const { Runtime } = Cu.import('resource://qbrt/modules/Runtime.jsm', {});
+const { Services } = Cu.import('resource://gre/modules/Services.jsm', {});
 
-window.addEventListener('load', event => {
-  const browser = document.getElementById('content');
-  const url = window.arguments[0];
+let Accelerators = {
+  isMac: Services.appinfo.OS === 'Darwin',
+  keys: {
+    i: 73,
+    r: 82,
+    f5: 116,
+  },
+  Reload: function(event) {
+    const f5 = event.keyCode === this.keys.f5;
+    if (f5) {
+      return true;
+    }
+    if (this.isMac) {
+      // `Cmd + R`.
+      const cmdR = event.metaKey && event.keyCode === this.keys.r;
+      return cmdR;
+    }
+    // `Ctrl + Shift + R`.
+    const ctrlShiftR = event.ctrlKey && event.keyCode === this.keys.r;
+    return ctrlShiftR;
+  },
+  HardReload: function(event) {
+    // `Shift + F5`.
+    const shiftF5 = event.shiftKey && event.keyCode === this.keys.f5;
+    if (shiftF5) {
+      return true;
+    }
+    if (this.isMac) {
+      // `Cmd + Shift + R`.
+      const cmdShiftR = event.metaKey && event.shiftKey &&
+        event.keyCode === this.keys.r;
```

Comment:
It's fine to break up this long line (and the ones below), but note that this repo's linter configuration allows lines to extend to a maximum length of 120 characters, so you could put this all on one line.